components/feature/contextmenu/src/test/java/mozilla/components/feature/contextmenu/ContextMenuCandidateTest.kt

```diff
@@ -237,120 +260,122 @@ class ContextMenuCandidateTest {
-        assertTrue(sessionManager.sessions[1].private)
-        assertEquals("https://firefox.com", sessionManager.sessions[1].url)
+        assertEquals(2, store.state.tabs.size)
+        assertTrue(store.state.tabs.last().content.private)
+        assertEquals("https://firefox.com", store.state.tabs.last().content.url)
     }
 
     @Test
     fun `Candidate "Save image"`() {
-        val saveImage = ContextMenuCandidate.createSaveImageCandidate(testContext)
+        val store = BrowserStore()
+        val sessionManager = spy(SessionManager(mock(), store))
+        doReturn(mock<EngineSession>()).`when`(sessionManager).getOrCreateEngineSession(any())
+        sessionManager.add(Session("https://www.mozilla.org", private = true))
+
+        val saveImage = ContextMenuCandidate.createSaveImageCandidate(
+            testContext,
+            ContextMenuUseCases(sessionManager))
 
         // showFor
 
         assertFalse(saveImage.showFor(
-            Session("https://www.mozilla.org"),
+            createTab("https://www.mozilla.org"),
             HitResult.UNKNOWN("https://www.mozilla.org")))
 
         assertFalse(saveImage.showFor(
-            Session("https://www.mozilla.org", private = true),
+            createTab("https://www.mozilla.org", private = true),
             HitResult.UNKNOWN("https://www.mozilla.org")))
 
         assertTrue(saveImage.showFor(
-            Session("https://www.mozilla.org"),
+            createTab("https://www.mozilla.org"),
             HitResult.IMAGE_SRC("https://www.mozilla.org", "https://www.mozilla.org")))
 
         assertTrue(saveImage.showFor(
-            Session("https://www.mozilla.org"),
+            createTab("https://www.mozilla.org"),
             HitResult.IMAGE("https://www.mozilla.org")))
 
         assertFalse(saveImage.showFor(
-            Session("https://www.mozilla.org"),
+            createTab("https://www.mozilla.org"),
             HitResult.VIDEO("https://www.mozilla.org")))
 
         // action
 
-        val session = Session("https://www.mozilla.org")
-
-        assertTrue(session.download.isConsumed())
+        assertNull(store.state.tabs.first().content.download)
 
-        saveImage.action.invoke(session,
+        saveImage.action.invoke(
+            store.state.tabs.first(),
             HitResult.IMAGE_SRC("https://www.mozilla.org/media/img/logos/firefox/logo-quantum.9c5e96634f92.png",
                 "https://firefox.com"))
 
-        assertFalse(session.download.isConsumed())
-
-        var imageDownload: Download? = null
-        session.download.consume { download ->
-            imageDownload = download
-            true
-        }
-
+        assertNotNull(store.state.tabs.first().content.download)
         assertEquals(
             "https://www.mozilla.org/media/img/logos/firefox/logo-quantum.9c5e96634f92.png",
-            imageDownload!!.url)
+            store.state.tabs.first().content.download!!.url)
     }
 
+    /*
```

Comment:
wow lots of tests to update. What about the ones you commented out? Did you still want to update those or file a follow-up? (happy to help)

Reply:
Oh, oops. I thought I had all of them fixed. Will look at that today.